docs/reference/api/platform/Error.md

```diff
@@ -1,14 +1,144 @@
+
+Note that the error functions output the error message or the file name in debug and develop builds only.
+
+Below is an example of terminal output the `MBED_ERROR1()` call created. Note that file name capture is disabled by default.
+
+```
+++ MbedOS Error Info ++
+Error Status: 0x80000110 Code: 272 Entity: 0
+Error Message: System type error
+Location: 0xab59
+File:main.cpp+222
+Error Value: 0xdeaddead
+Current Thread: Id: 0x200020e0 Entry: 0xcb31 StackSize: 0x1000 StackMem: 0x200010e0 SP: 0x20002020
+-- MbedOS Error Info --
+```
+
+### Constructing error status values in your implementation
+
+Mbed OS provides necessary functions and macros for implementations to construct error status values. There are few ways you can construct error status values. 
+
+If you know the module reporting the error you can use the **MBED_MAKE_ERROR()** macro to construct an error status with the module information. For example, if you want to report  an unsupported configuration from serial driver you may construct the error status as follows to capture the module information along with specific error code. In the below example, we are constructing an error status value with error code set to MBED_ERROR_CODE_CONFIG_UNSUPPORTED from serial driver indicated by module info set to MBED_MODULE_DRIVER_SERIAL.
+
+```mbed_error_status_t error = MBED_MAKE_ERROR(MBED_MODULE_DRIVER_SERIAL, MBED_ERROR_CODE_CONFIG_UNSUPPORTED)```
+
+There may be scenarios when the module might have called an API exported from other modules like protocol stacks but have received an error status in return. In those cases, it doesn't know which of the lower layers raised the error but we may still want to report the error without the module information. In those cases, it can use the predefined error status such as MBED_ERROR_CONFIG_UNSUPPORTED with the module value already set to MBED_MODULE_UNKNOWN.
+
+This is equivalent to defining an error status with MODULE_UNKNOWN as below. Although, using predefined values like MBED_ERROR_CONFIG_UNSUPPORTED makes it more convenient and easier to read the implementation.
+
+```mbed_error_status_t error = MBED_MAKE_ERROR(MBED_MODULE_UNKNOWN, MBED_ERROR_CODE_CONFIG_UNSUPPORTED)```
+
+### Error history
+
+Error handling implementation in Mbed OS also keeps track of previous errors in the system. This feature is called **Error history** and is configurable using the configuration value **MBED_CONF_ERROR_HIST_SIZE**.
+
+MBED_CONF_ERROR_HIST_SIZE configures the number of previous errors the system keeps in its error history. You can disable the history feature by setting MBED_CONF_ERROR_HIST_SIZE to 0 if required. By default, it keeps track of last four errors. Irrespective of whether error history is enabled or not, the system always records the first and last error happened in the system. See
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/mbed__error_8h_source.html) 
```

Comment:
I'd move the link to not be inline with the text

Reply:
Will fix